### PR TITLE
Simplify doNarrow, MainNavScreen and fix inconsistencies

### PR DESCRIPTION
**Pull request by borisyankov**
Some parts of the code were not updated to latest approaches.

* Extract doNarrow to an action, less passing of the function around
* Extract SideDrawer to new file
* Simplify MainNavBar and remove old and unused Navigation component
* Remove the manual fetchMessage calls in few places and replace with doNarrow

### CODE REVIEW

**Comment by smarx** [comment]
*Automated message from Dropbox CLA bot*

@borisyankov, it looks like you've already signed the Dropbox CLA. Thanks!

File: src/message/messagesActions.js [hunk 1]
```diff
@@ -73,27 +81,31 @@ export const fetchMessages = (
...
+        ...(numBefore ? { older: true } : {}),
+        ...(numAfter ? { newer: true } : {}),
+      }),
+    );
     dispatch(backgroundFetchMessages(auth, anchor, numBefore, numAfter, narrow, useFirstUnread));
   };
 
-export const updateMessageFlags = (auth, messageIds, op, flag) =>
-  async (dispatch) => {
-    await messagesFlags(auth, messageIds, op, flag);
```
Comment (neerajwahi) [inline src/message/messagesActions.js]: Where are we calling this now? We need to call it to tell the API that a message has been read

Comment (borisyankov) [inline src/message/messagesActions.js]: It is extracted in the queueMarkAsRead. This is just code that was forgotten to be removed.

Comment (neerajwahi) [inline src/message/messagesActions.js]: :+1:

**Pull request merged**